File: test/interface/golden/lib/js/__tests__/product/SubTypeParameter_spec.js
'use strict';

var AesonSpec        = require("bs-aeson-spec/lib/js/src/AesonSpec.js");
var SubTypeParameter = require("../../product/SubTypeParameter.js");

AesonSpec.sampleGoldenAndServerSpec((function (param) {
        return SubTypeParameter.decodeSubTypeParameter(AesonSpec.decodeIntWithResult, AesonSpec.decodeIntWithResult, AesonSpec.decodeIntWithResult, param);
      }), (function (param) {
        return SubTypeParameter.encodeSubTypeParameter((function (prim) {
                      return prim;
                    }), (function (prim) {
                      return prim;
                    }), (function (prim) {
                      return prim;
                    }), param);
      }), "subTypeParameter", "http://localhost:8081/SubTypeParameter/SubTypeParameter", "golden/product/SubTypeParameter");

/*  Not a pure module */

File: test/interface/golden/__tests__/product-servant/SubTypeParameter_spec.ml
let () =
  AesonSpec.sampleGoldenAndServerSpec
    (SubTypeParameter.decodeSubTypeParameter AesonSpec.decodeIntWithResult AesonSpec.decodeIntWithResult AesonSpec.decodeIntWithResult)
    (SubTypeParameter.encodeSubTypeParameter Aeson.Encode.int Aeson.Encode.int Aeson.Encode.int)
    "subTypeParameter"
    "http://localhost:8081/SubTypeParameter/SubTypeParameter"
    "golden/product/SubTypeParameter";
